o1vm/src/pickles/lookup_prover.rs

```diff
@@ -0,0 +1,226 @@
+use ark_ff::{One, PrimeField, Zero};
+use ark_poly::{univariate::DensePolynomial, Evaluations, Polynomial, Radix2EvaluationDomain};
+use kimchi::{circuits::domains::EvaluationDomains, curve::KimchiCurve, plonk_sponge::FrSponge};
+use mina_poseidon::FqSponge;
+use o1_utils::ExtendedDensePolynomial;
+use poly_commitment::{commitment::absorb_commitment, ipa::SRS, OpenProof, SRS as _};
+//TODO Parralelize
```

Comment:
This is done in a follow-up. However, it has been built on top of other PRs, and we cannot simply cherry-pick the changes as it creates conflicts. For the sake of simplicity and to move on, we will accept and fix it later.